PNA.mdk

```diff
@@ -902,39 +903,24 @@ A PNA implementation need not support both `pna_implementation` and
 ## Tables with add-on-miss capability {#sec-add-on-miss}
 
 PNA defines the `add_on_miss` table property.  If the value of this
-property is `true` for a table `t`, the P4 developer is allowed to
-define a default action for `t` that calls the `add_entry` extern
-function.
-
-When `t.apply()` is invoked, `t`'s lookup key is constructed, and the
-entries of the table are searched.  If there is no match, i.e. the
-lookup results in a miss, `t`'s default action is executed.  So far,
-this is all standard behavior as defined in the P4~16~ language
-specification.
-
-If `t`'s default action makes a call to `add_entry`, it causes a new
-entry to be added to the table with the same key that was just looked
-up and resulted in a miss, and the action name and action parameters
-specified by the parameters of the call to the `add_entry` extern
-function.  Thus, future packets that invoke `t.apply()` with the same
-lookup key will get a match and invoke the specified action (until and
-unless this new table entry is removed).  The new table entry will be
-matchable when the next packet is processed that invoked `t.apply()`.
-
-Some PNA implementations may allow the control plane software to add,
-modify, and delete entries of such a table, but any entries added via
-the `add_entry` function do not require the control plane software to
-be involved in any way.  Other PNA implementations may choose not to
-support control plane modification of the entries of an add-on-miss
-table.
-
-It is expected that PNA implementations will be able to sustain
-`add_entry` calls at a large fraction of their line rate, but it need
-not be at the same packet rate supported for processing packets that
-do not call `add_entry`.
+property is `true` for a table, the P4 developer is allowed to define
+a default action for the table that calls the `add_entry` extern
+function.  `add_entry` adds a new entry to the table whose default
+action calls the `add_entry` function.  The new entry will have the
+same key that was just looked up.
+
+The control plane API is still allowed to add, modify, and delete
+entries of such a table, but any entries added via the `add_entry`
+function do not require the control plane software to be involved in
+any way.  It is expected that PNA implementations will be able to
+sustain `add_entry` calls at a large fraction of their line rate, but
+it need not be at the same packet rate supported for processing
+packets that do not call `add_entry`.  The new table entry will be
+matchable when the next packet is processed that applies this table.
 
 ~Begin P4Example
-[INCLUDE=pna.p4:add_entry_extern_function]
+extern bool add_entry<T>(string action_name,
+                         in T action_params);
```

Comment:
Same here.  The INCLUDE Madoko directive is more recent, and easier to keep up to date with the pna.p4 file if its contents change in the future.